src/FSharpLint.Core/Rules/Conventions/AvoidSinglePipeOperator.fs

```diff
@@ -37,16 +37,53 @@ let runner (args: AstNodeRuleParams) =
         | _ ->
             Array.empty
 
+    let rec checkInnerExpressions (expr: SynExpr) =
+        match expr with
+        | SynExpr.App(_exprAtomicFlag, _isInfix, funcExpr, _argExpr, _range) ->
+            checkExpr funcExpr
+        | SynExpr.IfThenElse(_, thenExpr, elseExpr, _, _, _, _) ->
+            let checkThen = checkInnerExpressions thenExpr
+
+            match elseExpr with
+            | Some elseExpression ->
+                checkThen
+                |> Array.append (checkInnerExpressions elseExpression)
+            | None -> checkThen
+        | SynExpr.ArrayOrListComputed(_, expr, _) ->
+            checkInnerExpressions expr
+        | SynExpr.Sequential (_, _, firstExpression, secondExpression, _) ->
+            let result = checkInnerExpressions firstExpression
+            Array.append result (checkInnerExpressions secondExpression)
+        | SynExpr.Paren (innerExpression, _, _, _) ->
+            checkInnerExpressions innerExpression
+        | SynExpr.While (_, _, innerExpression, _) ->
+            checkInnerExpressions innerExpression
+        | SynExpr.For (_, _, _, _, _, _, _, innerExpression, _) ->
+            checkInnerExpressions innerExpression
+        | SynExpr.ForEach (_, _, _, _, _, _, innerExpression, _) ->
+            checkInnerExpressions innerExpression
+        | SynExpr.Match (_, _, clauses, _, _) ->
+            clauses
+            |> List.map (fun (SynMatchClause (_, _, clause, _, _, _)) -> checkInnerExpressions clause)
+            |> List.toArray
+            |> Array.concat
+        | SynExpr.Do (innerExpression, _) -> checkInnerExpressions innerExpression
+        | SynExpr.TryWith (tryExpression, withCases, _, _, _, _) ->
+            withCases
+            |> List.map (fun (SynMatchClause (_, _, withCase, _, _, _)) -> checkInnerExpressions withCase)
+            |> List.toArray
+            |> Array.concat
+            |> Array.append (checkInnerExpressions tryExpression)
+        | SynExpr.TryFinally (tryExpression, finallyExpr, _, _, _, _) ->
+            checkInnerExpressions finallyExpr
+            |> Array.append (checkInnerExpressions tryExpression)
+        | _ ->
+            Array.empty
+
     let error =
         match args.AstNode with
         | AstNode.Binding (SynBinding(_synAcc, _synBinding, _mustInline, _isMut, _synAttribs, _preXmlDoc, _synValData, _headPat, _synBindingRet, synExpr, _range, _debugPointAtBinding, _)) ->
```

Comment:
Is there a reason why only bindings are checked? What happens if pipeline operator is used in expression that is not a part of binding?
Like
```
module Foo

-1.0 |> abs
```

Maybe if you match `AstNode.Expression` instead, that big `checkInnerExpressions` function will be unnecessary?

Reply:
sure, let's do this, but let's add a test first (and one that doesn't generate errors or warnings, because `abs` doesn't return unit, so it would generate a warning AFAIU)